src/include/migraphx/heap.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_HEAP_HPP
+#define MIGRAPHX_GUARD_RTGLIB_HEAP_HPP
+
+#include <thread>
+#include <cmath>
+#include <algorithm>
+#include <vector>
+#include <cassert>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+template <class T, class Compare>
+struct heap
```

Comment:
This should probably be called `heap_vector` since this only works with `vector`.

Reply:
fixed